src/utils/itk.ts

```diff
@@ -0,0 +1,216 @@
+import { logError } from "@/utils/log";
+import { IGeoJSPoint, IToolConfiguration } from "@/store/model";
+import IOTypes from "itk/IOTypes";
+import runPipelineBrowser from "itk/runPipelineBrowser";
+
+function geoJSCoordinatesTo2DITKMesh(coordinates: IGeoJSPoint[], map: any) {
+  const pointsArray = new Float32Array(coordinates.length * 2);
+  for (let i = 0; i < coordinates.length; ++i) {
+    const displayCoordinates = map.gcsToDisplay(coordinates[i]);
+    pointsArray[2 * i] = displayCoordinates.x;
+    pointsArray[2 * i + 1] = displayCoordinates.y;
+  }
+
+  return {
+    meshType: {
+      dimension: 2,
+      pointComponentType: "float"
+    },
+    dimension: 2,
+    numberOfPoints: coordinates.length,
+    points: pointsArray,
+    numberOfPointPixels: 0,
+    pointData: new Uint32Array(),
+    numberOfCells: 0,
+    cells: new Uint32Array(),
+    numberOfCellPixels: 0,
+    cellData: new Uint32Array(),
+    cellBufferSize: 0
+  };
+}
+
+function runItkPipelineWrapper(
+  pipelineName: string,
+  image: Uint8Array,
+  otherArgs: { path: string; data: any; type: any }[]
+): Promise<any> {
+  return new Promise((resolve, reject) => {
+    const args: string[] = [
+      "/inimage.png",
+      ...otherArgs.map(arg => arg.path),
+      "/out.json"
+    ];
+    const pipelineInputs = [
+      {
+        path: "/inimage.png",
+        data: image,
+        type: IOTypes.Binary
+      },
+      ...otherArgs
+    ];
+    const pipelineOutputs = [
+      {
+        path: "out.json",
+        type: IOTypes.Text
+      }
+    ];
+    return runPipelineBrowser(
+      null,
+      pipelineName,
+      args,
+      pipelineOutputs,
+      pipelineInputs
+    )
+      .then(({ outputs, webWorker }: { outputs: any; webWorker: Worker }) => {
+        if (!outputs || outputs.length !== 1 || outputs[0].data === "") {
+          webWorker.terminate();
+          logError(`Pipeline didn't return a value`);
+          reject();
+        } else {
+          const result = JSON.parse(outputs[0].data);
+          webWorker.terminate();
+          resolve(result);
+        }
+      })
+      .catch((error: any) => {
+        logError(`Failed to run ITK pipeline:
+${error.message}
```

Comment:
indentation

Reply:
This is a literal string, if I add indentation, it will be included in the string
Do you mean that the error message also needs indentation? or I should reformat it?